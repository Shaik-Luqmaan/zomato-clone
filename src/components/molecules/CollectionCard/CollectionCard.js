import React from 'react'
import { makeStyles } from "@material-ui/core/styles";
import { Grid } from "@material-ui/core";
import Typography from "../../atoms/Typography/Typography";
import ArrowRightIcon from "@material-ui/icons/ArrowRight";

const useStyles = makeStyles({
    root: {
        borderRadius: "0.6rem",
        height: "320px",
        width: "280px",
        backgroundSize: "cover",
        backgroundPosition: "center",
        color: "white",
        cursor: "pointer",
      },
      arrow: {
        position: "relative",
        top: "8px",
      },
      large: {
        height: "300px",
        width: "100%"
      },
      content: {
        padding: "15px",
      },
      
})

const CollectionCard = ({
  text,
  placesCount,
  img,
  ...props
})=>{

  const classes = useStyles();
  let rootClass = classes.root;

  if (props.size === "large") {
    rootClass += " " + classes.large;
  }
  return (
    <Grid
      container
      alignItems="flex-end"
      className={rootClass}
      style={{
        backgroundImage: `linear-gradient(rgba(0,0,0, 0.1), rgb(0, 0, 0, .6)), url(${img})`,
      }}
      data-testid="main"
      {...props}
    >
      <Grid item className={classes.content}>
        <Typography data-testid="text" variant="h6" >
          {text}
        </Typography>
        <Typography  variant="subtitle2" data-testid="place">
          {placesCount} 
          <ArrowRightIcon className={classes.arrow} fontSize="small" />
        </Typography>
      </Grid>
    </Grid>
  )
}

export default CollectionCard
