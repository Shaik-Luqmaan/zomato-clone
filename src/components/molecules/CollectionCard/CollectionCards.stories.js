import React from 'react';
import CollectionCard from './CollectionCard'
import collection from '../../pictures/collection_1.jpg'

export default {
    title: 'molecules/CollectionCard',
    component: CollectionCard
}

const Template = (args) => <CollectionCard {...args} />

export const ViewPageCollection = Template.bind({});
ViewPageCollection.args={
    text: "Most Searched 2020",
    placesCount : '28 Places',
    img: collection,
}

export const RoutePageCollection = Template.bind({});
RoutePageCollection.args={
    text: "Most Searched 2020",
    placesCount : '28 Places',
    img: collection,
    size: 'large'
}