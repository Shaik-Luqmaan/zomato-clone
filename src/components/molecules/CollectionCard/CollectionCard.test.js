import React from "react";
import ReactDOM from "react-dom";
import { render, cleanup } from "@testing-library/react";
import "@testing-library/jest-dom";
import CollectionCard from './CollectionCard';
import collection from '../../pictures/collection_1.jpg'
import renderer from 'react-test-renderer';

afterEach(cleanup);

it("matches snapshot",()=>{
    const args={
        text: "Most Searched 2020",
        placesCount : '28 Places',
        img: collection,
    };

    const tree = renderer.create(<CollectionCard {...args}/>).toJSON();
    expect(tree).toMatchSnapshot();
})

it("renders component without crashing", ()=>{
    const div = document.createElement("div");
    ReactDOM.render(<CollectionCard />, div)
})

it("renders component correctly",()=>{
    const args={
        text: "Most Searched 2020",
        placesCount : '28 Places',
        img: collection,
    };
    const{getByTestId} = render(<CollectionCard {...args} /> )
    expect(getByTestId("text")).toHaveTextContent(args.text);
    expect(getByTestId("place")).toHaveTextContent(args.placesCount);
})

it("matches snapshot 2",()=>{
    const tree = renderer.create(<CollectionCard/>).toJSON();
    expect(tree).toMatchSnapshot();
})