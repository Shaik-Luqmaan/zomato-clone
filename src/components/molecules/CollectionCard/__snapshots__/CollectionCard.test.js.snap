// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<div
  className="MuiGrid-root makeStyles-root-1 MuiGrid-container MuiGrid-align-items-xs-flex-end"
  data-testid="main"
  style={
    Object {
      "backgroundImage": "linear-gradient(rgba(0,0,0, 0.1), rgb(0, 0, 0, .6)), url(collection_1.jpg)",
    }
  }
>
  <div
    className="MuiGrid-root makeStyles-content-4 MuiGrid-item"
  >
    <h6
      className="MuiTypography-root MuiTypography-h6"
      data-testid="text"
    >
      Most Searched 2020
    </h6>
    <h6
      className="MuiTypography-root MuiTypography-subtitle2"
      data-testid="place"
    >
      28 Places
      <svg
        aria-hidden={true}
        className="MuiSvgIcon-root makeStyles-arrow-2 MuiSvgIcon-fontSizeSmall"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M10 17l5-5-5-5v10z"
        />
      </svg>
    </h6>
  </div>
</div>
`;

exports[`matches snapshot 2 1`] = `
<div
  className="MuiGrid-root makeStyles-root-1 MuiGrid-container MuiGrid-align-items-xs-flex-end"
  data-testid="main"
  style={
    Object {
      "backgroundImage": "linear-gradient(rgba(0,0,0, 0.1), rgb(0, 0, 0, .6)), url(undefined)",
    }
  }
>
  <div
    className="MuiGrid-root makeStyles-content-4 MuiGrid-item"
  >
    <h6
      className="MuiTypography-root MuiTypography-h6"
      data-testid="text"
    />
    <h6
      className="MuiTypography-root MuiTypography-subtitle2"
      data-testid="place"
    >
      <svg
        aria-hidden={true}
        className="MuiSvgIcon-root makeStyles-arrow-2 MuiSvgIcon-fontSizeSmall"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M10 17l5-5-5-5v10z"
        />
      </svg>
    </h6>
  </div>
</div>
`;
